File: server/java/integration-tests/src/test/java/com/metaobjects/integration/api/ApiContractScenarios.java
package com.metaobjects.integration.api;

import java.util.List;
import java.util.Map;

/**
 * Typed records mirroring the YAML shape under
 * {@code fixtures/api-contract-conformance/scenarios/}. See the corpus
 * README for the supported {@code expect.body.*} vocabulary; we model it as
 * a permissive {@code Map} so this code stays decoupled from whichever
 * subset of assertion keys a given scenario happens to use.
 *
 * <p>Mirror of {@code ApiContractScenarios.kt} in
 * {@code integration-tests-kotlin}.</p>
 */
final class ApiContractScenarios {
    private ApiContractScenarios() {}

    /** A single HTTP request inside a scenario. */
    record ApiRequest(
        String id,
        String method,                  // GET | POST | PATCH | PUT | DELETE
        String path,                    // includes the query string
        Object body,                    // JSON-shaped (Map | List | scalar) or null
        int expectStatus,
        Map<String, Object> expectBody  // nullable
    ) {}

    /** A scenario file (one {@code .yaml} under {@code scenarios/}). */
    record ApiScenario(
        String name,
        String description,
        String sourcePath,
        boolean truncate,               // true when setup.truncate is set
        List<ApiRequest> requests
    ) {}
}

File: server/java/integration-tests-kotlin/src/test/kotlin/com/metaobjects/integration/kotlin/api/ApiContractScenarios.kt
package com.metaobjects.integration.kotlin.api

/**
 * Typed records mirroring the YAML shape under
 * `fixtures/api-contract-conformance/scenarios/`. See the corpus README for
 * the supported `expect.body.*` vocabulary; we model it as a permissive
 * map so this code stays decoupled from whichever subset of assertion keys
 * a given scenario happens to use.
 */
object ApiContractScenarios {

    /** A single HTTP request inside a scenario. */
    data class ApiRequest(
        val id: String,
        val method: String,           // GET | POST | PATCH | PUT | DELETE
        val path: String,             // includes the query string
        val body: Any? = null,        // JSON-shaped (Map | List | scalar) or null
        val expectStatus: Int,
        val expectBody: Map<String, Any?>? = null,
    )

    /** A scenario file (one `.yaml` under `scenarios/`). */
    data class ApiScenario(
        val name: String,
        val description: String,
        val sourcePath: String,
        val truncate: Boolean,        // true when setup.truncate is set
        val requests: List<ApiRequest>,
    )
}
